### ISSUE TITLE
Record not shown after transition in a relationship many to many

### ISSUE THREAD

**Issue by HenryVonfire** [emberjs/data#4673]
Hi, I have a relationship many to many with the models `user, address` with an intermediate model (user-to-address). When a user wants to delete an `address`, I store the records `address` and `user-to-address` in a service, remove `user-to-address` record from the relationship and set them as deleted with `deleteRecord`, allowing the user to undo the action and recover them.

To undo the action, I get those records from the service, rollback their attributes with `rollBackAttributes` and add them to the user using `pushObject` on the relationship.

If I delete an address and press the `undo` button it works perfectly, the address disappears and reappears again. But, if I delete an address, go back to the previous route, enter again on `User 1` and press the `undo` button, the address isn't shown.

I've tested that the same implementation works with a 1 to many relationship. Here's an [ember-twiddle](https://ember-twiddle.com/d5796dcf0b18207e352c7cbe9b51a3c6) that reflects this behaviour.

tl;dr
in ember-twiddle:
1. click on `users` link
2. click on `User 1` link
3. click on the `x` button next to Street 1
4. click on `back` button
5. click again on `User 1` link
6. click on `undo` button

Omitting steps 4 and 5 the example works as expected.

**Comment by bmac** [comment]
I spent some time trying to debug this today. I didn't come to anything conclusive but I noticed that calling `flushCanonical` on the relationships updates the value so it will render correctly. I'm out of time for now but hopefully this bit of info might help someone else looking at this issue.

**Comment by runspired** [comment]
@HenryVonfire is this still an issue on latest master?

**Comment by HenryVonfire** [comment]
```
DEBUG: -------------------------------
DEBUG: Ember      : 3.0.0
DEBUG: Ember Data : 3.0.2
```
@runspired I've created a new project with the above versions and now it works :smile: Sorry for keeping the issue opened so long but I'm glad to see that the problem has been fixed. Thanks for your effort guys! :heart:
